test/encode.spec.ts

```diff
@@ -80,4 +82,61 @@ describe('encode', () => {
       expect(output[i]).to.be.an.instanceOf(Uint8Array)
     }
   })
+
+  it('should not encode message data that is too long', async () => {
+    const input = [new Uint8Array(MAX_DATA_LENGTH + 1)] // Create a buffer that exceeds the max length
+    try {
+      await pipe(
+        input,
+        (source) => lp.encode(source),
+        async (source) => all(source)
+      )
+      throw new Error('Expected error not thrown') // Fail the test if the error isn't thrown
+    } catch (err: any) {
+      expect(err).to.be.an.instanceof(InvalidDataLengthError)
+      expect(err.code).to.equal('ERR_MSG_DATA_TOO_LONG') // Check the error code
+    }
+  })
+
+  it('should throw an error if message data exceeds custom maxDataLength', async () => {
+    const customMaxDataLength = 512 // Set a custom max data length
+    const input = [new Uint8Array(customMaxDataLength + 1)] // Create a buffer larger than the custom limit
+
+    const options = { maxDataLength: customMaxDataLength } // Set maxDataLength in options
+
+    try {
+      await pipe(
+        input,
+        (source) => lp.encode(source, options),
+        async (source) => all(source)
+      )
+      throw new Error('Expected error not thrown') // Fail the test if the error isn't thrown
+    } catch (err: any) {
+      expect(err).to.be.an.instanceof(InvalidDataLengthError)
+      expect(err.code).to.equal('ERR_MSG_DATA_TOO_LONG') // Check the error code
+    }
+  })
+
+  it('should encode data within custom maxDataLength', async () => {
+    const customMaxDataLength = 512 // Set a custom max data length
+    const input = [new Uint8Array(customMaxDataLength)] // Create a buffer within the allowed size
+
+    const options = { maxDataLength: customMaxDataLength } // Set maxDataLength in options
+    const output = await pipe(
+      input,
+      (source) => lp.encode(source, options),
+      async (source) => all(source)
+    )
+
+    let inputIndex = 0
+
+    for (let i = 0; i < output.length; i += 2, inputIndex++) {
+      const prefix = output[i]
+      const data = output[i + 1]
+
+      const length = varint.decode(prefix)
+      expect(length).to.equal(data.length)
+      expect(data).to.deep.equal(input[inputIndex])
```

Comment:
You can be slightly more expressive with other [matchers](https://www.chaijs.com/api/bdd/) (plus the ones [bundled with aegir](https://github.com/ipfs/aegir/blob/main/utils/chai.js#L29-L33)):

```suggestion
      expect(data).to.haveLengthOf(length)
      expect(data).to.equalBytes(input[inputIndex])
```